### PR TITLE
Port the parameterised release workflow here

### PR DESCRIPTION
**Pull request by squaremo**
.. from image-automation-controller. It is exactly the same except for the name of the controller, used in various capacities.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -49,8 +52,8 @@ jobs:
           file: ./Dockerfile
           platforms: linux/amd64
           tags: |
-            ghcr.io/fluxcd/image-reflector-controller:${{ steps.prep.outputs.VERSION }}
-            docker.io/fluxcd/image-reflector-controller:${{ steps.prep.outputs.VERSION }}
+            ghcr.io/fluxcd/${{ env.CONTROLLER }}:${{ steps.prep.outputs.VERSION }}
```
Comment (stefanprodan) [inline .github/workflows/release.yml:55]: I think you can use `${{ github.event.repository.name }}` instead of hardcoding the controller name.

Comment (squaremo) [reply .github/workflows/release.yml:55]: I'll try it!

**Comment by stefanprodan** [comment]
I can’t comment on this line FROM golang:1.15 as builder but it should be golang-alpine

**Comment by squaremo** [comment]
> I can’t comment on this line FROM golang:1.15 as builder but it should be FROM golang:1.15-alpine as builder

How would you expect this to fail -- failed `make docker-build`, or unusable image?

**Review by stefanprodan — APPROVED** [review]
LGTM 💯

File: .github/workflows/release.yml (hunk 1 above)
Comment (squaremo) [reply .github/workflows/release.yml:55]: (yes it works!)

**Pull request merged**